includes/multi-currency/class-frontend-currencies.php

```diff
@@ -0,0 +1,155 @@
+<?php
+/**
+ * WooCommerce Payments Multi Currency Frontend Currencies
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Multi_Currency;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class that formats Multi Currency currencies on the frontend.
+ */
+class Frontend_Currencies {
+	/**
+	 * Multi-Currency instance.
+	 *
+	 * @var Multi_Currency
+	 */
+	protected $multi_currency;
+
+	/**
+	 * Multi-Currency currency formatting map.
+	 *
+	 * @var array
+	 */
+	protected $currency_settings = [];
+
+	/**
+	 * Constructor.
+	 *
+	 * @param Multi_Currency $multi_currency The Multi_Currency instance.
+	 */
+	public function __construct( Multi_Currency $multi_currency ) {
+		$this->multi_currency = $multi_currency;
+
+		$this->load_locale_data();
+
+		// Currency hooks.
+		add_filter( 'woocommerce_currency', [ $this, 'get_current_currency_code' ], 50 );
+		add_filter( 'wc_get_price_decimals', [ $this, 'get_current_currency_decimals' ], 50 );
+		add_filter( 'wc_get_price_decimal_separator', [ $this, 'get_current_currency_decimal_separator' ], 50 );
+		add_filter( 'wc_get_price_thousand_separator', [ $this, 'get_current_currency_thousand_separator' ], 50 );
+		add_filter( 'woocommerce_price_format', [ $this, 'get_current_currency_format' ], 50 );
+	}
+
+	/**
+	 * Returns the currency code to be used by WooCommerce.
+	 *
+	 * @return string The code of the currency to be used.
+	 */
+	public function get_current_currency_code() {
+		return $this->multi_currency->get_selected_currency()->get_code();
+	}
+
+	/**
+	 * Returns the number of decimals to be used by WooCommerce.
+	 *
+	 * @return int The number of decimals.
+	 */
+	public function get_current_currency_decimals() {
+		$currency_code = $this->multi_currency->get_selected_currency()->get_code();
```

Comment:
Little nitpick/question here. Should we use `get_current_currency_code` here and in the following getters? To avoid reusing the same code `$this->multi_currency->get_selected_currency()->get_code()`.

Reply:
I'm fine either way 😄 

I didn't reuse it at first because that method is mainly intended to be used by the `woocommerce_currency` filter and it could change over time.

But, right now all it does is wrap `$this->multi_currency->get_selected_currency()->get_code()` so we might as well just use it, even if it might cause a bigger refactor in the future in case we need to do something else when running the `woocommerce_currency` filter.

Does that make sense? How do you think we should approach using this method?

Reply:
That said, it would be good to rename these methods to tie them a bit more with the hook they're attached to if we treat them as such. E.g. instead of `get_current_currency_code` we should name it `get_woocommerce_currency` or something along those lines.

Reply:
If we probably change this method in the future, better to leave as it is. just looks like a handy getter even it's used in the hook. And agree to rename the methods attached to hooks.

Reply:
Agreed. I've updated the names in https://github.com/Automattic/woocommerce-payments/pull/1860/commits/f5472150d5de5c08b383ab1ce92e8346784937fe. What do you think of them?